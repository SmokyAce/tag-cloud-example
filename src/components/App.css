.App {
    font-family: Lucida Console, Courier, monospace;
    text-align: center;
    display: flex;
    flex-direction: column;
    height: 100vh;
}

.text-center {
    text-align: center
}
