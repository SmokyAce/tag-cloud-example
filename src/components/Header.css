
.header {
    margin: 50px 50px 0;
}

.header h1 {
    font-size: 3em;
    margin: 0;
}
