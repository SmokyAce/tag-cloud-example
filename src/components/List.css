#list .subheader {
    list-style-type: none;
    position: relative;
    font-size: 1.1em;
    left: -30px;
}

#list-item {
    padding: 5px 0;
}
