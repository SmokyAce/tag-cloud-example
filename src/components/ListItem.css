#list .subheader {
    list-style-type: none;
    font-size: 1.1em;
    left: -40px;
    position: relative;
}

#list-item {
    padding: 5px 0;
}
