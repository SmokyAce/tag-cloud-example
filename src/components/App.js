import React, { lazy, Suspense } from 'react';
import {
  BrowserRouter as Router,
  Route,
  Redirect,
  Switch
} from 'react-router-dom';
import Spinner from './Spinner';
import './App.css';

const HomePage = lazy(() => import('./HomePage'));
const TagPage = lazy(() => import('./TagPage'));
const NotFoundPage = lazy(() => import('./NotFoundPage'));

const App = () => {
  return (
    <div className='App'>
      <Router>
        <Suspense fallback={<Spinner />}>
          <Switch>
            <Route exact path='/(home)?' component={() => <HomePage />} />
            <Route
              path='/home/:tagId+'
              component={props => <TagPage {...props} />}
            />
            <Route component={() => <NotFoundPage />} />
            <Redirect to='/' />
          </Switch>
        </Suspense>
      </Router>
    </div>
  );
};

export default App;
