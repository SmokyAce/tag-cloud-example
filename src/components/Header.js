import React from 'react';
import PropTypes from 'prop-types';
import './Header.css';

const Header = ({ children, ...props }) => {
  return (
    <div className='header' {...props}>
      <h1>{children}</h1>
    </div>
  );
};

Header.propTypes = {
  children: PropTypes.any
};

export default Header;
