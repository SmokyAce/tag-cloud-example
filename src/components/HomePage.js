import React from 'react';
import Header from './Header';
import TagCloud from './TagCloud';

const HomePage = props => {
  return (
    <React.Fragment>
      <Header>Tag Cloud</Header>
      <TagCloud {...props} />
    </React.Fragment>
  );
};

export default HomePage;
