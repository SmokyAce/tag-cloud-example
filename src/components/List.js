import React from 'react';
import PropTypes from 'prop-types';
import './List.css';

export const List = ({ children, ...props }) => {
  return (
    <ul id='list' {...props}>
      {children}
    </ul>
  );
};

List.propTypes = {
  children: PropTypes.any.isRequired
};

const className = (isTrue, addClassName) => {
  return isTrue ? addClassName : ''
};

export const ListItem = ({ content, subheader = false, ...props }) => {
  return (
    <li id='list-item' className={className(subheader, 'subheader')} {...props}>
      {content}
    </li>
  );
};

ListItem.propTypes = {
  content: PropTypes.any.isRequired,
  subheader: PropTypes.bool
};
