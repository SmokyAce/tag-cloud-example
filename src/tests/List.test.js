/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import { List, ListItem } from '../components/List';

describe('<List />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <List>
        <ListItem content='Subheader' subheader />
        <ListItem content='Item 1' />
        <ListItem content='Item 2' />
      </List>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const snapshot = TestRenderer.create(
      <List>
        <ListItem content='Subheader' subheader />
        <ListItem content='Item 1' />
        <ListItem content='Item 2' />
      </List>
    ).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
