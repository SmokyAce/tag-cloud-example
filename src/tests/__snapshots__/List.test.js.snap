// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders correctly 1`] = `
<ul
  id="list"
>
  <li
    className="subheader"
    id="list-item"
  >
    Subheader
  </li>
  <li
    className=""
    id="list-item"
  >
    Item 1
  </li>
  <li
    className=""
    id="list-item"
  >
    Item 2
  </li>
</ul>
`;
